### PR TITLE
[CDBS-53] Bump dependencies of the project.

### PR DESCRIPTION
**Pull request by ksaric**
https://github.com/input-output-hk/cardano-db-sync/issues/54

This is a start. Moving to dependencies where we can use Shelley.
Run the tests, the ones not connected to the DB pass, need to check why it constantly shouts that PGPASSFILE is not present when it is.
Also, run in sync with the node, seems to work across EBB:
```
PGPASSFILE=config/pgpass db-sync-node/bin/cardano-db-sync \
    --config config/mainnet-config.yaml \
    --genesis-file ../cardano-node/configuration/mainnet-genesis.json \
    --socket-path ../cardano-node/state-node-mainnet/node.socket \
    --schema-dir schema/
[db-sync-node:Info:4] [2020-04-08 10:42:25.24 UTC] NetworkMagic: RequiresNoMagic 764824073
[db-sync-node:Info:4] [2020-04-08 10:42:25.49 UTC] Initial genesis distribution present and correct
[db-sync-node:Info:4] [2020-04-08 10:42:25.49 UTC] Total genesis supply of Ada: 31112484745.000000
[db-sync-node:Info:4] [2020-04-08 10:42:25.49 UTC] localInitiatorNetworkApplication: connecting to node via "../cardano-node/state-node-mainnet/node.socket"
[db-sync-node.Handshake:Info:38] [2020-04-08 10:42:25.49 UTC] {"event":"Send MsgProposeVersions (fromList [(NodeToClientV_1,TInt 764824073)])","kind":"LocalHandshakeTrace","bearer":"ConnectionId {localAddress = LocalAddress {getFilePath = \"\"}, remoteAddress = LocalAddress {getFilePath = \"state-node-mainnet/node.socket\"}}"}
[db-sync-node.Handshake:Info:38] [2020-04-08 10:42:25.50 UTC] {"event":"Recv MsgAcceptVersion NodeToClientV_1 (TInt 764824073)","kind":"LocalHandshakeTrace","bearer":"ConnectionId {localAddress = LocalAddress {getFilePath = \"\"}, remoteAddress = LocalAddress {getFilePath = \"state-node-mainnet/node.socket\"}}"}
[db-sync-node:Info:41] [2020-04-08 10:42:25.50 UTC] Starting chainSyncClient
[db-sync-node:Info:41] [2020-04-08 10:42:25.56 UTC] Cardano.Db tip is at slot 30111, block 30097
[db-sync-node:Info:45] [2020-04-08 10:42:25.56 UTC] Running DB thread
[db-sync-node:Info:45] [2020-04-08 10:42:25.84 UTC] No rollback required: chain tip slot is 30111
[db-sync-node:Info:45] [2020-04-08 10:42:38.19 UTC] insertABlock: slot 35000, block 34982, hash 5ffb04538391bf4ba30acb5c6d911c51b389fafb15a5feed21b141a07a57d9d0
[db-sync-node:Info:45] [2020-04-08 10:42:46.74 UTC] insertABlock: slot 40000, block 39976, hash f6440713d5a17d0b020e9246f542a4c0b611ed4065acfd1dfe9b868b986a564f
[db-sync-node:Info:45] [2020-04-08 10:42:54.90 UTC] insertABOBBoundary: epoch 2 hash d46adef46f0760f7b89bd03926a0a631aded69b30379ca0404a0f68ebd2936dc
[db-sync-node:Info:45] [2020-04-08 10:42:56.31 UTC] insertABlock: slot 45000, block 44976, hash a8f2d5f14df667e45e25808dfc8f038078888c86ebf1a994d52cc1594b865335
[db-sync-node:Info:45] [2020-04-08 10:43:01.18 UTC] insertABlock: slot 50000, block 49975, hash f53b48aa3e8b9a6dbe5e6ca1562f912b53ba4d4be837374ac47aebe8c928de4c
[db-sync-node:Info:45] [2020-04-08 10:43:07.53 UTC] insertABlock: slot 55000, block 54975, hash 7e99cb45a96e0a4497d85281ab6eb679843712121ac9ae6898b2920b379634a8
[db-sync-node:Info:45] [2020-04-08 10:43:16.31 UTC] insertABlock: slot 60000, block 59975, hash 8fe1c7df25000a8123e65c24fc61172ea00ad7a5239a285a8b272dde17d92bde
[db-sync-node:Info:45] [2020-04-08 10:43:20.18 UTC] insertABOBBoundary: epoch 3 hash 7dd0c73651016e0706ee5f18e0b35bc6579e683b25db5e596e09eca9346f0065
[db-sync-node:Info:45] [2020-04-08 10:43:20.30 UTC] insertABlock: slot 65000, block 64974, hash 2f444ea3970f0ac36d64e04abb430fe1618c5a9338590ac4f2bc890f27ecc0a5
[db-sync-node:Info:45] [2020-04-08 10:43:26.63 UTC] insertABlock: slot 70000, block 69973, hash cc0707892fbc2eac429f4d22c2b69ce773e5e2bae5afeeaeec16cebc7da8e717
[db-sync-node:Info:45] [2020-04-08 10:43:36.02 UTC] insertABlock: slot 75000, block 74971, hash 6dda16d41d396462b54cb4104ddf0c933dc86a1683d5bb53cec532472fc95040
[db-sync-node:Info:45] [2020-04-08 10:43:40.30 UTC] insertABlock: slot 80000, block 79971, hash 1a39cbeb8ea750be80eec939781ebeb6a01f5438f6c090164a0d7c09024dac67
[db-sync-node:Info:45] [2020-04-08 10:43:51.46 UTC] insertABlock: slot 85000, block 84971, hash 8abb057703eaf09cb494b9f46321cedf7c6b894852f062df843ba77ddb3f4c0a
[db-sync-node:Info:45] [2020-04-08 10:43:52.81 UTC] insertABOBBoundary: epoch 4 hash 8360a8d832a4e59c91efb21576a0aa5f3cf185e067da729cdc15331204128685
[db-sync-node:Info:45] [2020-04-08 10:43:56.90 UTC] insertABlock: slot 90000, block 89971, hash 3fbabec9ad9268be4a81bb6f82085509d2347f4d8d70628cc41cc126038c506d
[db-sync-node:Info:45] [2020-04-08 10:44:06.68 UTC] insertABlock: slot 95000, block 94971, hash f87c8473273ddfc4f1ba0bfb6148f9a533bf7b9e8914652d14f56c19a60e58ee
[db-sync-node:Info:45] [2020-04-08 10:44:11.27 UTC] insertABlock: slot 100000, block 99971, hash 566101de2f6a51b1fe8a88cc4d2caf8a0e277bef52a72fd4a4a03bb11608db1e
[db-sync-node:Info:45] [2020-04-08 10:44:16.73 UTC] insertABlock: slot 105000, block 104971, hash 855407bc35273c7cd9a5181c779a8f42b864071c76dcb5169d202782b0cbaeae
[db-sync-node:Info:45] [2020-04-08 10:44:22.33 UTC] insertABOBBoundary: epoch 5 hash 9021389e9fe671b9fd47ea15ff1e17ab37870e30a3053de8f5e60649f6ef0ec8
[db-sync-node:Info:45] [2020-04-08 10:44:25.74 UTC] insertABlock: slot 110000, block 109971, hash 8961519245585c8a0a1a33a6509edbcde0356e10bba03b0d1b581bd25ef249b9
[db-sync-node:Info:45] [2020-04-08 10:44:30.58 UTC] insertABlock: slot 115000, block 114969, hash f8beec2a0de323344ae0a8bb901613cc53886208957c76bac6889f7b1102ae29
[db-sync-node:Info:45] [2020-04-08 10:44:36.70 UTC] insertABlock: slot 120000, block 119969, hash 6c088daee7af052c003bb799c5b51d4777216ac18b42a4515aafedca74e5219d
[db-sync-node:Info:45] [2020-04-08 10:44:44.71 UTC] insertABlock: slot 125000, block 124969, hash 68315ec6e7f0bdeacace0e3ed90df011fd8a511443fafe8c8cf6ed2e1048e96b
[db-sync-node:Info:45] [2020-04-08 10:44:49.63 UTC] insertABOBBoundary: epoch 6 hash 6ca7fd6d27b48e980c9c8f6499c8f783317ae9686f3b8a458125825c7ce32e54
[db-sync-node:Info:45] [2020-04-08 10:44:50.14 UTC] insertABlock: slot 130000, block 129969, hash f178283253f3ea4db6c10c74490544733e833eb7d593f8bbd378697ea23653dc
[db-sync-node:Info:45] [2020-04-08 10:44:54.59 UTC] insertABlock: slot 135000, block 134969, hash 10c82e97715c81576581a1b43665db4f1e8bc1431742ffb8f2113745d929a5b7
[db-sync-node:Info:45] [2020-04-08 10:45:03.53 UTC] insertABlock: slot 140000, block 139969, hash f81932e01887aba38a33d5d0149bf50cff79e05c435083ddcd9ba77dca9b48cd
[db-sync-node:Info:45] [2020-04-08 10:45:07.99 UTC] insertABlock: slot 145000, block 144969, hash 39e1a67f4b1794b6a0bbbe51781ff49e9315e33c1e71722a54a6b460957629ce
[db-sync-node:Info:45] [2020-04-08 10:45:14.49 UTC] insertABlock: slot 150000, block 149969, hash 11f17d90c7a2d4f690904b07a27b3a96682a44952383981267d010f33dc5743a
[db-sync-node:Info:45] [2020-04-08 10:45:15.24 UTC] insertABOBBoundary: epoch 7 hash 8313ccac635027eb2150c27f417c00541371f0a40cbc5fc714dc6c3b0e93ac19
[db-sync-node:Info:45] [2020-04-08 10:45:21.87 UTC] insertABlock: slot 155000, block 154969, hash 2a6aafd8f0e5afc3b8dec453a4c4c5d919ff3b87d8533fbb473ea9dfd6cecdd3
[db-sync-node:Info:45] [2020-04-08 10:45:26.24 UTC] insertABlock: slot 160000, block 159969, hash c1688121aaabb9228e7b18f2721002b22093575d18c5faa68d0a5329cdd19001
[db-sync-node:Info:45] [2020-04-08 10:45:34.91 UTC] insertABlock: slot 165000, block 164969, hash 5443d18bf25a06979d09f51f5d87193922b005c6929cf7149a90e62e0086d581
[db-sync-node:Info:45] [2020-04-08 10:45:39.32 UTC] insertABlock: slot 170000, block 169968, hash f93695a364ff27120abbedd6b7d5b70ad4fe861401efe7a79f56cdc3cf5b9578
[db-sync-node:Info:45] [2020-04-08 10:45:41.32 UTC] insertABOBBoundary: epoch 8 hash e96a7fa23688a0d0f68bfdacddd1a1976f1c26724c484a6e35099d77d1acf186
[db-sync-node:Info:45] [2020-04-08 10:45:45.89 UTC] insertABlock: slot 175000, block 174968, hash dd4910a2636ed867ad91931d07643dca49d7eda2aa1bc56f708b3b3a33f923bc
[db-sync-node:Info:45] [2020-04-08 10:45:49.71 UTC] insertABlock: slot 180000, block 179968, hash cd0deea6c2fd854b056377643db9084e73a77a7a466c5b374ba4dd94d9502a8f
[db-sync-node:Info:45] [2020-04-08 10:45:59.51 UTC] insertABlock: slot 185000, block 184968, hash 377ddb8d47d8aaf1c12d3735c89f5937d5d7af9e06787ec85051e41dfca7751b
[db-sync-node:Info:45] [2020-04-08 10:46:03.63 UTC] insertABlock: slot 190000, block 189968, hash 2e6cdd6d550ba470bbc2174e4e65c94d6a02bf4c01a0d47ebff04f81c8105684
[db-sync-node:Info:45] [2020-04-08 10:46:11.76 UTC] insertABOBBoundary: epoch 9 hash b86032b4c1a3b9e5569fccf1573016d6f35efd06a4f1002a4bb97058941dc103
[db-sync-node:Info:45] [2020-04-08 10:46:12.16 UTC] insertABlock: slot 195000, block 194968, hash 7d75bcebf4a0a8ec211294f09a5eea9cf2e89a8c9df5bc007b5bec6b4008ea64
[db-sync-node:Info:45] [2020-04-08 10:46:16.23 UTC] insertABlock: slot 200000, block 199968, hash e4b605ac05d3701415246421c5b62d4d439301b92c3bac6f41361e3971d193da
[db-sync-node:Info:45] [2020-04-08 10:46:21.97 UTC] insertABlock: slot 205000, block 204968, hash 987c83cfee58d31180a576e41fb9ab2b8f77f71bfd1cf67b7b80d550d5bd45a4
[db-sync-node:Info:45] [2020-04-08 10:46:29.90 UTC] insertABlock: slot 210000, block 209968, hash 4f4826943917e10a479585852da3c964fa1cf74c68eb39637d0fac74a343d5e1
[db-sync-node:Info:45] [2020-04-08 10:46:37.64 UTC] insertABlock: slot 215000, block 214968, hash f58dbea85298fe737d221d5c37362606ec47862e14b1c8f42bbd0578b23247a1
[db-sync-node:Info:45] [2020-04-08 10:46:38.46 UTC] insertABOBBoundary: epoch 10 hash af82845f28b968c49eeb5b9cdb902af357435fe2a2c45dfb9ac7196adf9c1470
[db-sync-node:Info:45] [2020-04-08 10:46:41.56 UTC] insertABlock: slot 220000, block 219968, hash a519b6099df20583b4716c221eba79f0b8569a563cd19dc6e828eb65754dcbda
[db-sync-node:Info:45] [2020-04-08 10:46:47.34 UTC] insertABlock: slot 225000, block 224968, hash 02d3454215adbe52e314c4a1ba516b2559f6852a644f1f00f6134e3efac7314b
[db-sync-node:Info:45] [2020-04-08 10:46:55.53 UTC] insertABlock: slot 230000, block 229968, hash 64e759dbca0652b22d342b5dae13dd21503a4983072d8d9beaaa384f95b9446e
[db-sync-node:Info:45] [2020-04-08 10:46:59.62 UTC] insertABlock: slot 235000, block 234968, hash 2ac38a8e0f2bdc13a5db11754e3105cdf244f55b872269853eaa645b3a2514c4
[db-sync-node:Info:45] [2020-04-08 10:47:03.22 UTC] insertABOBBoundary: epoch 11 hash 0f27259d20d9c8cc17780b672ec1efeb224acb25ca84d17ec4689325b06d804a
[db-sync-node:Info:45] [2020-04-08 10:47:08.12 UTC] insertABlock: slot 240000, block 239968, hash 2fc85d276ce608c59f47ef0133ebac1ccd48f0032dbfbaf2f27038b0e2d0407f
[db-sync-node:Info:45] [2020-04-08 10:47:12.09 UTC] insertABlock: slot 245000, block 244967, hash 94671d4ce7f88c59b9f4ae94ade15f841316d06b091ff9797bc4b67f480f5953
^C[db-sync-node:Error:48] [2020-04-08 10:47:12.65 UTC] recvMsgRollForward: AsyncCancelled
[db-sync-node:Error:45] [2020-04-08 10:47:12.65 UTC] runDBThread: AsyncCancelled
```

### CODE REVIEW

**Comment by erikd** [comment]
The environment variable `PGPASSFILE` is missing. You need:
```
export PGPASSFILE=$(pwd)/config/pgpass
cabal test all -j1
```

File: cardano-db-sync/src/Cardano/DbSync.hs [hunk 1]
```diff
@@ -441,6 +441,8 @@ chainSyncClient trce metrics latestPoints currentTip actionQueue =
 logProtocolMagic :: Trace IO Text -> Crypto.ProtocolMagic -> IO ()
 logProtocolMagic tracer pm =
   liftIO . logInfo tracer $ mconcat
-    [ "NetworkMagic: ", textShow (Crypto.getRequiresNetworkMagic pm), " "
+    [ "NetworkMagic: "
+    , textShow (Crypto.getRequiresNetworkMagic pm)
+    , " "
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync.hs:446]: Why? That change is purely whitespace?

Comment (ksaric) [reply cardano-db-sync/src/Cardano/DbSync.hs:446]: Yes, this is my preference. I see this as more clear to read.

Comment (erikd) [reply cardano-db-sync/src/Cardano/DbSync.hs:446]: I see these extra new lines as un-necessary. That is my preference.

File: cardano-db-sync/src/Cardano/DbSync/Util.hs [hunk 2]
```diff
@@ -139,13 +140,17 @@ pointToSlotHash (Point x) =
     Origin -> Nothing
     At blk -> Just (Ledger.SlotNumber . unSlotNo $ Point.blockPointSlot blk, unByronHash $ Point.blockPointHash blk)
 
-renderAbstractHash :: ByteArrayAccess bin => bin -> Text
-renderAbstractHash =
+renderByteArrayToText :: ByteArrayAccess bin => bin -> Text
+renderByteArrayToText =
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync/Util.hs:144]: There is no function `renderByteArrayToString` so 'ToText` is unnecesary.

The convention in this code base is `renderX :: X -> Type`.

File: cardano-db-sync/src/Cardano/DbSync/Tracing/ToObjectOrphans.hs [hunk 3]
```diff
@@ -23,10 +23,6 @@ import           Ouroboros.Network.Protocol.ChainSync.Type (ChainSync)
 instance Transformable Text IO (TraceSendRecv (ChainSync ByronBlock (Tip ByronBlock))) where
   trTransformer = defaultTextTransformer
 
-instance HasPrivacyAnnotation (TraceSendRecv (ChainSync ByronBlock (Tip ByronBlock)))
-
-instance HasSeverityAnnotation (TraceSendRecv (ChainSync ByronBlock (Tip ByronBlock)))
-
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync/Tracing/ToObjectOrphans.hs:29]: :100:

**Comment by ksaric** [comment]
Used `./nix/regenerate.sh` (again), not sure why the build is failing.

File: cardano-db-sync/src/Cardano/DbSync/Util.hs [hunk 4]
```diff
@@ -18,7 +18,7 @@ module Cardano.DbSync.Util
   , logException
   , mkSlotLeader
   , pointToSlotHash
-  , renderByteArrayToText
+  , renderByteArray
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync/Util.hs:21]: :+1:

**Comment by erikd** [comment]
The "[CDBS-53] Fixed review comments." part of the message is completely redundant.

**Pull request merged**